offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -220,11 +224,87 @@ class ProductsDbConnector {
     return productItems
   }
 
+  /**
+   * A product is archived by setting it inactive
+   * @param product The product that needs to be archived
+   * @since 1.0.0
+   * @throws life.qbic.business.exceptions.DatabaseQueryException
+   */
+  @Override
+  void archive(Product product) throws DatabaseQueryException {
+    Connection connection = provider.connect()
+
+    connection.withCloseable {
+      def statement = connection.prepareStatement(Queries.ARCHIVE_PRODUCT)
+      statement.setString(1, product.productId.toString())
+      statement.execute()
+    }
+  }
+
+  /**
+   * Fetches a product from the database
+   * @param productId The product id of the product to be fetched
+   * @return returns an optional that contains the product if it has been found
+   * @since 1.0.0
+   * @throws life.qbic.business.exceptions.DatabaseQueryException is thrown when any technical interaction with the data source fails
+   */
+  @Override
+  Optional<Product> fetch(ProductId productId) throws DatabaseQueryException {
+    Connection connection = provider.connect()
+    String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
+    Optional<Product> product = Optional.empty()
+
+    connection.withCloseable {
+      PreparedStatement preparedStatement = it.prepareStatement(query)
+      preparedStatement.setString(1, productId.identifier.toString())
+      ResultSet result = preparedStatement.executeQuery()
+
+      while (result.next()) {
+        product = Optional.of(rowResultToProduct(SqlExtensions.toRowResult(result)))
+      }
+    }
+    return product
+  }
+
+  /**
+   * Stores a product in the database
+   * @param product The product that needs to be stored
+   * @since 1.0.0
+   * @throws DatabaseQueryException if any technical interaction with the data source fails
+   * @throws ProductExistsException if the product already exists in the data source
+   */
+  @Override
+  void store(Product product) throws DatabaseQueryException, ProductExistsException {
+    Connection connection = provider.connect()
+
+    connection.withCloseable {
+      PreparedStatement preparedStatement = it.prepareStatement(Queries.INSERT_PRODUCT)
+      preparedStatement.setString(1, getProductType(product))
+      preparedStatement.setString(2, product.description)
+      preparedStatement.setString(3, product.productName)
+      preparedStatement.setDouble(4, product.unitPrice)
+      preparedStatement.setString(5, product.unit.value)
+      preparedStatement.setString(6, product.productId.toString())
+
+      preparedStatement.execute()
+    }
+  }
+
   /**
    * Class that encapsulates the available SQL queries.
    */
   private static class Queries {
 
+    /**
+     * Query for inserting a product.
+     */
+    final static String INSERT_PRODUCT = "INSERT INTO product (category, description, productName, unitPrice, unit, productId) VALUES (?, ?, ?, ?, ?, ?)"
+
+    /**
+     * Inactivate a product
```

Comment:
Nit: 
```suggestion
     * Query for Inactivation of a product
```

Reply:
I think we do not query for it but just do it.